app/models/piece.rb

```diff
@@ -2,11 +2,61 @@ class Piece < ApplicationRecord
 
   belongs_to :game
   
+  # Determines if a piece can be captured
+	def capturable?(x, y)
+	  (piece_present_at?(x, y)) && !is_same_color?(x, y)
+	end
+	
+	# Captures present piece if is capturable (changes db)
+	def capture!(x, y)
+	  if piece_present_at?(x, y)
+		  if capturable?(x, y)
+		    update_captured_piece!(x, y)
+		    move_to!(x, y)
+		    return true
+      end
+		else
+		  move_to!(x, y)
+		  return true
+		end
+	end
+  
+	# Changes captured piece attributes to reflect capture (changes db)
+	def update_captured_piece!(x, y)
+	  self.present_piece(x, y).update_attributes(position_x: nil, position_y: nil,
+	                                              state: 'captured', dead: true)
```

Comment:
why do we have both a state and a boolean for dead?

Reply:
Thanks for you assistance on this one, you have a sharp eye.